NullPointerException: Cannot invoke "String.equals(Object)" because "reqGroup" is null

## What version of OpenRewrite are you using?
I am using
- org.openrewrite:rewrite-maven 8.36.0
- org.openrewrite.recipe:rewrite-spring 5.20.0

## How are you running OpenRewrite?
command line

## What is the smallest, simplest way to reproduce the problem?
don't have any atm.
We have a few projects that use several plugins that are not use by most projects. Those get the exception.
The issue is in [not checking for `null`](https://github.com/openrewrite/rewrite/blob/v8.36.0/rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java#L407) like in [this place](https://github.com/openrewrite/rewrite/blob/v8.36.0/rewrite-maven/src/main/java/org/openrewrite/maven/MavenVisitor.java#L131) where `null` is checked.

I assume there are transitive plugins in plugin management using `org.apache.maven.plugins` plugins and not setting the _groupId_ explicitly.

## What did you expect to see?
no NullPointerException

## What is the full stack trace of any errors you encountered?
```
[ERROR] Failed to execute goal org.openrewrite.maven:rewrite-maven-plugin:5.41.0:run (default-cli) on project our-internal-project: Execution default-cli of goal org.openrewrite.maven:rewrite-maven-plugin:5.41.0:run failed: Error while visiting pom.xml: java.lang.NullPointerException: Cannot invoke "String.equals(Object)" because "reqGroup" is null
[ERROR]   org.openrewrite.maven.MavenVisitor.findPlugin(MavenVisitor.java:407)
[ERROR]   org.openrewrite.maven.MavenVisitor.findManagedPlugin(MavenVisitor.java:400)
[ERROR]   org.openrewrite.maven.RemoveRedundantDependencyVersions$1.visitTag(RemoveRedundantDependencyVersions.java:201)
[ERROR]   org.openrewrite.maven.RemoveRedundantDependencyVersions$1.visitTag(RemoveRedundantDependencyVersions.java:163)
[ERROR]   org.openrewrite.xml.tree.Xml$Tag.acceptXml(Xml.java:454)
[ERROR]   org.openrewrite.xml.tree.Xml.accept(Xml.java:50)
[ERROR]   org.openrewrite.TreeVisitor.visit(TreeVisitor.java:245)
[ERROR]   org.openrewrite.TreeVisitor.visitAndCast(TreeVisitor.java:317)
[ERROR]   org.openrewrite.xml.XmlVisitor.lambda$visitTag$2(XmlVisitor.java:91)
[ERROR]   org.openrewrite.internal.ListUtils.map(ListUtils.java:176)
[ERROR]   org.openrewrite.xml.XmlVisitor.visitTag(XmlVisitor.java:91)
[ERROR]   org.openrewrite.maven.MavenIsoVisitor.visitTag(MavenIsoVisitor.java:38)
[ERROR]   org.openrewrite.maven.RemoveRedundantDependencyVersions$1.visitTag(RemoveRedundantDependencyVersions.java:219)
[ERROR]   org.openrewrite.maven.RemoveRedundantDependencyVersions$1.visitTag(RemoveRedundantDependencyVersions.java:163)
[ERROR]   org.openrewrite.xml.tree.Xml$Tag.acceptXml(Xml.java:454)
[ERROR]   org.openrewrite.xml.tree.Xml.accept(Xml.java:50)

```

## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?
I could, but the fix seems so small that a regular contributor or team member can also fix it quickly.

--- a/rewrite-maven/src/main/java/org/openrewrite/maven/RemoveRedundantDependencyVersions.java
+++ b/rewrite-maven/src/main/java/org/openrewrite/maven/RemoveRedundantDependencyVersions.java
@@ -364,12 +364,9 @@ public class RemoveRedundantDependencyVersions extends Recipe {
         };
     }
 
-    private static @Nullable String getManagedPluginVersion(ResolvedPom resolvedPom, @Nullable String groupId, String artifactId) {
+    private static @Nullable String getManagedPluginVersion(ResolvedPom resolvedPom, String groupId, String artifactId) {
         for (Plugin p : ListUtils.concatAll(resolvedPom.getPluginManagement(), resolvedPom.getRequested().getPluginManagement())) {
-            if (Objects.equals(
-                    Optional.ofNullable(p.getGroupId()).orElse("org.apache.maven.plugins"),
-                    Optional.ofNullable(groupId).orElse("org.apache.maven.plugins")) &&
-                Objects.equals(p.getArtifactId(), artifactId)) {
+            if (Objects.equals(p.getGroupId(), groupId) && Objects.equals(p.getArtifactId(), artifactId)) {
                 return resolvedPom.getValue(p.getVersion());
             }
         }

--- a/rewrite-maven/src/main/java/org/openrewrite/maven/tree/Plugin.java
+++ b/rewrite-maven/src/main/java/org/openrewrite/maven/tree/Plugin.java
@@ -21,6 +21,8 @@ import com.fasterxml.jackson.databind.JsonNode;
 import lombok.AccessLevel;
 import lombok.Value;
 import lombok.experimental.FieldDefaults;
+
+import org.jspecify.annotations.NonNull;
 import org.jspecify.annotations.Nullable;
 import org.openrewrite.maven.internal.MavenXmlMapper;
 
@@ -33,6 +35,7 @@ public class Plugin {
     // default value as per https://maven.apache.org/xsd/maven-4.0.0.xsd
     public static final String PLUGIN_DEFAULT_GROUPID = "org.apache.maven.plugins";
 
+    @Nullable
     String groupId;
     String artifactId;
 
@@ -50,6 +53,11 @@ public class Plugin {
 
     List<Dependency> dependencies;
     List<Execution> executions;
+
+    @NonNull
+    public String getGroupId() {
+        return groupId == null ? PLUGIN_DEFAULT_GROUPID : groupId;
+    }
 
     @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
     @Value

